CDATA handling in HTML changed in lxml parser with libxml2 2.9.12

After upgrading the system libxml2 to 2.9.12 (or 2.9.11; 2.9.10 is the previous working version I have here), the two following tests fail with lxml built against the system library:

```
FAILED tests/test_extra/test_soup_contains.py::TestSoupContains::test_contains_cdata_html - AssertionError: Lists differ: ['1', '2'] != ['1']
FAILED tests/test_extra/test_soup_contains_own.py::TestSoupContainsOwn::test_contains_own_cdata_html - AssertionError: Lists differ: ['1', '2']...
```

The cause seems to be a different representation of CDATA:

```
        soup       = <html><body><div id="1">Testing that <span id="2">&lt;![CDATA[that]]&gt;</span>contains works.</div></body>
</html>
```

(i.e. `&lt![CDATA[...` instead of `<!--[CDATA[...`)

Note that in order to reproduce you need to both upgrade libxml2 and build lxml against the new version. Binary wheels are statically linked to an old version of libxml2, so they do not reproduce the issue yet. For example, I have been able to reproduce it with tox after swapping the installed lxml version:
```
. .tox/py39/bin/activate
pip uninstall lxml
pip install lxml --no-binary lxml
```

I am also not sure whether this isn't a bug in libxml2 or lxml.

--- a/.pyspelling.yml
+++ b/.pyspelling.yml
@@ -21,7 +21,7 @@ matrix:
       ignores:
       - 'code, pre, a.magiclink, span.keys'
       - '.MathJax_Preview, .md-nav__link, .md-footer-custom-text, .md-source__repository, .headerlink, .md-icon'
-      - '.md-footer-social__link'
+      - '.md-social__link'
   - pyspelling.filters.url:
 
 - name: markdown

--- a/tests/test_extra/test_soup_contains.py
+++ b/tests/test_extra/test_soup_contains.py
@@ -144,7 +144,7 @@ class TestSoupContains(util.TestCase):
             flags=util.HTML
         )
 
-    def test_contains_cdata_html(self):
+    def test_contains_cdata_html5(self):
         """Test contains CDATA in HTML5."""
 
         markup = """
@@ -155,7 +155,40 @@ class TestSoupContains(util.TestCase):
             markup,
             'body *:-soup-contains("that")',
             ['1'],
-            flags=util.HTML
+            flags=util.HTML5
+        )
+
+    def test_contains_cdata_py_html(self):
+        """Test contains CDATA in Python HTML parser."""
+
+        markup = """
+        <body><div id="1">Testing that <span id="2"><![CDATA[that]]></span>contains works.</div></body>
+        """
+
+        self.assert_selector(
+            markup,
+            'body *:-soup-contains("that")',
+            ['1'],
+            flags=util.PYHTML
+        )
+
+    @util.skip_no_lxml
+    def test_contains_cdata_lxml_html(self):
+        """Test contains CDATA in `lxml` HTML parser."""
+
+        from lxml import etree
+        LIBXML_VER = etree.LIBXML_VERSION
+
+        markup = """
+        <body><div id="1">Testing that <span id="2"><![CDATA[that]]></span>contains works.</div></body>
+        """
+
+        results = ['1', '2'] if LIBXML_VER >= (2, 9, 11) else ['1']
+        self.assert_selector(
+            markup,
+            'body *:-soup-contains("that")',
+            results,
+            flags=util.LXML_HTML
         )
 
     def test_contains_cdata_xhtml(self):

--- a/tests/test_extra/test_soup_contains.py
+++ b/tests/test_extra/test_soup_contains.py
@@ -177,13 +177,13 @@ class TestSoupContains(util.TestCase):
         """Test contains CDATA in `lxml` HTML parser."""
 
         from lxml import etree
-        LIBXML_VER = etree.LIBXML_VERSION
+        libxml_ver = etree.LIBXML_VERSION
 
         markup = """
         <body><div id="1">Testing that <span id="2"><![CDATA[that]]></span>contains works.</div></body>
         """
 
-        results = ['1', '2'] if LIBXML_VER >= (2, 9, 11) else ['1']
+        results = ['1', '2'] if libxml_ver >= (2, 9, 11) else ['1']
         self.assert_selector(
             markup,
             'body *:-soup-contains("that")',

--- a/tests/test_extra/test_soup_contains_own.py
+++ b/tests/test_extra/test_soup_contains_own.py
@@ -35,7 +35,7 @@ class TestSoupContainsOwn(util.TestCase):
             flags=util.HTML
         )
 
-    def test_contains_own_cdata_html(self):
+    def test_contains_own_cdata_html5(self):
         """Test contains CDATA in HTML5."""
 
         markup = """
@@ -46,7 +46,40 @@ class TestSoupContainsOwn(util.TestCase):
             markup,
             'body *:-soup-contains-own("that")',
             ['1'],
-            flags=util.HTML
+            flags=util.HTML5
+        )
+
+    def test_contains_own_cdata_py_html(self):
+        """Test contains CDATA in Python HTML parser."""
+
+        markup = """
+        <body><div id="1">Testing that <span id="2"><![CDATA[that]]></span>contains works.</div></body>
+        """
+
+        self.assert_selector(
+            markup,
+            'body *:-soup-contains-own("that")',
+            ['1'],
+            flags=util.PYHTML
+        )
+
+    @util.skip_no_lxml
+    def test_contains_own_cdata_lxml_html(self):
+        """Test contains CDATA in `lxml` HTML."""
+
+        from lxml import etree
+        LIBXML_VER = etree.LIBXML_VERSION
+
+        markup = """
+        <body><div id="1">Testing that <span id="2"><![CDATA[that]]></span>contains works.</div></body>
+        """
+
+        results = ['1', '2'] if LIBXML_VER >= (2, 9, 11) else ['1']
+        self.assert_selector(
+            markup,
+            'body *:-soup-contains-own("that")',
+            results,
+            flags=util.LXML_HTML
         )
 
     def test_contains_own_cdata_xml(self):

--- a/tests/test_extra/test_soup_contains_own.py
+++ b/tests/test_extra/test_soup_contains_own.py
@@ -68,13 +68,13 @@ class TestSoupContainsOwn(util.TestCase):
         """Test contains CDATA in `lxml` HTML."""
 
         from lxml import etree
-        LIBXML_VER = etree.LIBXML_VERSION
+        libxml_ver = etree.LIBXML_VERSION
 
         markup = """
         <body><div id="1">Testing that <span id="2"><![CDATA[that]]></span>contains works.</div></body>
         """
 
-        results = ['1', '2'] if LIBXML_VER >= (2, 9, 11) else ['1']
+        results = ['1', '2'] if libxml_ver >= (2, 9, 11) else ['1']
         self.assert_selector(
             markup,
             'body *:-soup-contains-own("that")',

